core/src/test/java/com/sequenceiq/cloudbreak/core/flow2/cluster/verticalscale/CoreVerticalScaleServiceTest.java

```diff
@@ -0,0 +1,168 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.cloud.model.instance.InstanceStorageInfo;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterApi;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterModificationService;
+import com.sequenceiq.cloudbreak.converter.spi.InstanceMetaDataToCloudInstanceConverter;
+import com.sequenceiq.cloudbreak.core.bootstrap.service.host.ClusterHostServiceRunner;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterService;
+import com.sequenceiq.cloudbreak.service.stack.InstanceGroupService;
+import com.sequenceiq.cloudbreak.service.template.TemplateService;
+import com.sequenceiq.cloudbreak.template.model.ServiceComponent;
+
+@ExtendWith(MockitoExtension.class)
+public class CoreVerticalScaleServiceTest {
+
+    private static final String YARN_LOCAL_DIR = "yarn_nodemanager_local_dirs";
+
+    private static final String YARN_LOG_DIR = "yarn_nodemanager_log_dirs";
+
+    private static final String IMPALA_SCRATCH_DIR = "scratch_dirs";
+
+    private static final String IMPALA_DATACACHE_DIR = "datacache_dirs";
+
+    @Mock
+    private ClusterService clusterService;
+
+    @Mock
+    private InstanceGroupService instanceGroupService;
+
+    @Mock
+    private TemplateService templateService;
+
+    @Mock
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Mock
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Mock
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Mock
+    private ClusterHostServiceRunner clusterHostServiceRunner;
+
+    @InjectMocks
+    private CoreVerticalScaleService underTest;
+
+    @Mock
+    private StackDto stackDto;
+
+    @Mock
+    private ClusterApi clusterApi;
+
+    private ClusterModificationService clusterModificationService;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        doReturn(clusterApi).when(clusterApiConnectors).getConnector(stackDto);
+        clusterModificationService = mock(ClusterModificationService.class);
```

Comment:
Please also remove this one and annotate the field with `@Mock` instead.

Reply:
done